protocol/src/main/kotlin/tbdex/sdk/protocol/models/Rfq.kt

```diff
@@ -68,31 +95,117 @@ class Rfq private constructor(
     // TODO check that VCs satisfying PD are crypto verified
 
     try {
-      PresentationExchange.satisfiesPresentationDefinition(data.claims, requiredClaims)
+      PresentationExchange.satisfiesPresentationDefinition(
+        privateData?.claims ?: emptyList(),
+        requiredClaims
+      )
     } catch (e: Exception) {
       throw IllegalArgumentException("No matching claim for Offering requirements: ${requiredClaims.id}")
     }
   }
 
+  /**
+   * Verify the presence and integrity of all possible properties in Rfq.privateData
+   * @throws Exception if there are properties missing in Rfq.privateData or which do not match the corresponding
+   *                   hashed property in Rfq.data
+   */
+  fun verifyAllPrivateData() {
+    // Verify payin details
+    data.payin.paymentDetailsHash?.let {
+      verifyPayinDetailsHash()
+    }
+
+    // Verify payout details
+    data.payout.paymentDetailsHash?.let {
+      verifyPayoutDetailsHash()
+    }
+
+    // Verify claims
+    data.claimsHash?.let {
+      verifyClaimsHash()
+    }
+  }
+
+  /**
+   * Verify the integrity properties that are present in
+   * @throws Exception if there are properties present in Rfq.privateData which do not match the corresponding
+   *                   hashed property in Rfq.data
+   */
+  fun verifyPresentPrivateData() {
+    privateData ?: throw Error("privateData property is missing")
+
+    // Verify payin details
+    if (data.payin.paymentDetailsHash != null && privateData.payin?.paymentDetails != null) {
+      verifyPayinDetailsHash()
+    }
+
+    // Verify payout details
+    if (data.payout.paymentDetailsHash != null && privateData.payout?.paymentDetails != null) {
+      verifyPayoutDetailsHash()
+    }
+
+    // Verify claims
+    if (!data.claimsHash.isNullOrEmpty() && !privateData.claims.isNullOrEmpty()) {
+      verifyClaimsHash()
+    }
+  }
+
+  private fun verifyPayinDetailsHash() {
+    privateData ?: throw IllegalArgumentException("privateData property is missing")
```

Comment:
instead of all these `throws` you can actually do something like this

Reply:
```suggestion
    check(privateData != null) { "privateData property is missing" }
```

if `check` condition is false (i.e. privateData is null) this will throw `IllegalStateException`. i think you can also do `require(...) {...} ` which throws IllegalArgumentException

Reply:
good call, that looks way cleaner. done